miio/push_server.py

```diff
@@ -0,0 +1,430 @@
+import asyncio
+import calendar
+import datetime
+import logging
+import socket
+import struct
+from json import dumps
+from random import randint
+from typing import Any, Optional
+
+import attr
+
+from .device import Device
+from .protocol import Message, Utils
+
+_LOGGER = logging.getLogger(__name__)
+
+SERVER_PORT = 54321
+FAKE_DEVICE_ID = "120009025"
+FAKE_DEVICE_MODEL = "chuangmi.plug.v3"
+HELO_BYTES = bytes.fromhex(
+    "21310020ffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
+)
+
+
+@attr.s(auto_attribs=True)
+class EventInfo:
+    """Event info to register to the push server."""
```

Comment:
Please add docs to `command_extra`, `trigger_` to the docstring, as this is the main place where anyone using this would look for it.